patchy/api.py

```diff
@@ -275,14 +275,8 @@ def _process_function():
     six.exec_(new_code, func.__globals__, localz)
     new_func = localz['__patchy_freevars__']()
 
-    # Figure out how to get the Code object
-    if isinstance(new_func, (classmethod, staticmethod)):
```

Comment:
so the code in the new_source is never actually decorated with a classmethod or staticmethod - and it doesn't need to be (it seems)

Reply:
oh wait no - it's a python 2 thing